src/routes/reference/bufnmf.svx

```diff
@@ -9,4 +9,65 @@ tags:
     - spectral
 flair: reference
 category: Decompose
----
\ No newline at end of file
+---
+
+<script>
+    import IndentNote from '$lib/components/IndentNote.svelte';
+    import Image from '$lib/components/Image.svelte';
+    import Audio from '$lib/components/Audio.svelte';
+</script>
+
+## BufNMF
+
+BufNMF uses Non-Negative Matrix Factorisation (NMF) to decompose the magnitudes of a spectrogram into a number of components specified by the user<sup>1</sup>. Each component is represented by the combination of two elements: a spectral template (“basis”, plural: “bases”) and an amplitude envelope (“activation”). Activations and bases can be used in various ways, including to resynthesize the audio of a single decomposed component. NMF is a popular technique in signal processing research for things like source separation and transcription<sup>2</sup>.
+
+<IndentNote type='warn'>
+
+For an example of audio decomposition using NMF, see the [Decomposition with NMF](/overviews/bufnmf) Overview.
+
+</IndentNote>
+
+<Image
+src="/overviews/bufnmf/01_Bases_plot.png"
+label="Bases or 'Spectral Templates' of a decomposed drum loop. Each is a spectrum of magnitudes. (blue = kick, orange = snare, green = hi-hat)"
+/>
+
+<Image
+src="/overviews/bufnmf/00_resynth_buffer_with_activations.png"
+label="Activations from the same decomposed drum loop, overlaid on the decomposed audio components. (blue = kick, orange = snare, green = hi-hat)"
+/>
+
+NMF is a form of unsupervised machine learning: it begins either from seeded activations and bases or in a stochastic state (where the bases and activations of the components are initially random) and works iteratively, using stochastic gradient descent, trying to find a combination of components that, when summed together, minimize the difference from the original spectrum. In this implementation, the components are reconstructed by masking the original spectrum, such that they will always sum to yield the original sound.
+
+There is no single correct solution to this process. There are different ways of accounting for an spectrum in terms of some set of bases and activations. NMF tends to converge very quickly at first and then level out. Fewer iterations mean less processing, but also less predictable results.
+
+### Outputs
+
+BufNMF can return any or all of the following:
+* The bases of the components in the form of a magnitude spectrum for each.
+* The activations of the components in the form of an amplitude envelope for each.
+* An audio reconstruction of each component.
+
+### Different Modes
+
+Some additional options and flexibility can be found through combinations of the **basesMode** and **actMode** arguments.
+
+If either or both of these flags are set to 1, BufNMF expects to be supplied with the corresponding elements (activations or bases) to use as seeds for the decomposition, providing more guided results. It is possible to set both flags equal to 1.
```

Comment:
flags or parameters or attributes? what do ya reckon

Reply:
yeah, i guess i'm actually going to go with arguments since that's what the list down below is called and these will be in that list.